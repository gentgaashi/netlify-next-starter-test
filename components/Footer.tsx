import styled from 'styled-components';

const Sfooter = styled.footer`
  width: 100%;
  height: 100px;
  border-top: 1px solid #eaeaea;
  display: flex;
  justify-content: center;
  align-items: center;
`;
const Simg = styled.img`
  height: 1em;
  margin: 5px;
`;

export default function Footer() {
  return (
    <>
      <Sfooter>
        Made with{' '}
        <Simg src="/netliheart.svg" alt="Netlify Logo" /> for
        you
      </Sfooter>
    </>
  )
}
