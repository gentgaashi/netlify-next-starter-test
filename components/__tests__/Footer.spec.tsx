import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Footer from '../Footer';

let app: ShallowWrapper;

describe('With Enzyme', () => {
	beforeEach(() => {
		app = shallow(<Footer />);
	});

	it('to match snapshot', () => {
		expect(app).toMatchSnapshot();
	});

	it('App shows "Hi Next" in a <p> tag', () => {
		const ftr = app.find('Sfooter');
		expect(ftr).toBeTruthy();
		expect(ftr).toHaveStyleRule('display', 'flex');
		expect(ftr.find('Simg').prop('src')).toEqual('/netliheart.svg');
	});
});
