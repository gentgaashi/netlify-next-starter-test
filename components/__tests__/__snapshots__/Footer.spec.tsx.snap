// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme to match snapshot 1`] = `
<Fragment>
  <Sfooter>
    <p>
      Hi Next
    </p>
    Made with 
    <Simg
      alt="Netlify Logo"
      src="/netliheart.svg"
    />
     for you
  </Sfooter>
</Fragment>
`;
